nearly complete with find by author name in iter5

colehart opened the pull request.

BrianZanti commented on lib/library.rb:
```diff
@@ -0,0 +1,35 @@
+  end
+
+  def add_to_collection(book)
+    @books << book.shift
```
Instead of doing this, your add_book method in author should return a book object, not an array

BrianZanti commented on lib/author.rb:
```diff
@@ -1,3 +1,19 @@
+    @books << Book.new(author_first_name: @first_name,
+                       author_last_name: @last_name,
+                       title: title,
+                       publication_date: publication_date)
```
This should return the new Book, not the entire @books array

BrianZanti commented on lib/library.rb:
```diff
@@ -0,0 +1,35 @@
+  end
+
+  def card_catalogue
+    @books.sort_by!(&:author_last_name)
```
You don't want to use the ! here because this will change your array of books

BrianZanti commented on lib/library.rb:
```diff
@@ -0,0 +1,35 @@
+
+  def find_by_author(author_name)
+    name_ary = author_name.split(' ')
+    book_ary = @books.map do |book|
```
don't use data type in variable names

BrianZanti commented on lib/library.rb:
```diff
@@ -0,0 +1,35 @@
+    name_ary = author_name.split(' ')
+    book_ary = @books.map do |book|
+      book if book.author_first_name == name_ary[0]
+    end.compact
```
What you were looking for was find_all

colehart commented:
Thanks, Brian. The find_all was clutch. I’ve refactored the project and all is working now according to your feedback. One question I have about coverage being in the pull request:

I deleted it after the very first push to origin master when I realized it was getting pushed up, then added a .gitignore with coverage in there so it wouldn’t add every time. I’m pretty sure I removed it and git removed the coverage file, too, because when I hopped back on today and checked, it wasn’t on there.

So, my question is: if you mistakenly push coverage and then remove and git rm -r -f it and add coverage to a .gitignore, does it still show up in the pull request?

Thanks,
Cole

> On Apr 11, 2018, at 12:48 PM, BrianZanti <notifications@github.com> wrote:
> 
> @BrianZanti commented on this pull request.
> 
> In coverage/.last_run.json <https://github.com/turingschool-examples/denver_puplic_library/pull/20#discussion_r180860379>:
> 
> > @@ -1,5 +0,0 @@
> -{
> Don't push up your coverage! There are so many files in this PR!
> 
> —
> You are receiving this because you authored the thread.
> Reply to this email directly, view it on GitHub <https://github.com/turingschool-examples/denver_puplic_library/pull/20#pullrequestreview-111357471>, or mute the thread <https://github.com/notifications/unsubscribe-auth/AjEuCgaYKqZAdpH3PcSueUsSXCvcFA1Eks5tnk_7gaJpZM4TQZdo>.
>
